nototools/autofix_for_phase3.py

```diff
@@ -0,0 +1,270 @@
+
+def _check_version_info(version_info):
+  """ensure version info looks reasonable, for example:
+  'GOOG;noto-fonts:20170220:a8a215d2e889'.  Raise an exception
+  if it does not."""
+  m = _version_info_re.match(version_info)
+  if not m:
+    raise Exception('did not match version info "%s"' % version_info)
+  year = int(m.group(1))
+  month = int(m.group(2))
+  day = int(m.group(3))
+  commit_hash = m.group(4)
+  today = datetime.date.today()
+  if 2017 <= year:
+    try:
+      encoded_date = datetime.date(year, month, day)
+    except Exception as e:
+      raise Exception(
+          '%04d-%02d-%02d in %s is not a valid date' % (
+              year, month, day, version_info))
+    if encoded_date > today:
+      raise Exception('%s in %s is after the current date' % (
+          encoded_date, version_info))
+  else:
+    raise Exception('date in %s appears too far in the past' % version_info)
+
+
+def _check_autohint(script):
+  if script and not(
+      script in ['no-script'] or script in noto_data.HINTED_SCRIPTS):
+    raise Exception('not a hintable script: "%s"' % script)
+
+
+def autofix_fonts(font_names, dstdir, version_info, autohint, dry_run):
+  dstdir = tool_utils.ensure_dir_exists(dstdir)
+
+  _check_version_info(version_info)
+  _check_autohint(autohint)
+
+  font_names.sort()
+  print 'Processing\n  %s' % '\n  '.join(font_names)
+  print 'Dest dir: %s' % dstdir
+  if dry_run:
+    print '*** dry run %s***' % ('(autohint) ' if autohint else '')
+  for f in font_names:
+    fix_font(f, dstdir, version_info, autohint, dry_run)
+
+
+version_re = re.compile(r'Version (\d+\.\d{2,3})')
+def get_revision(font, fname):
+  if 'Armenian' in fname:
+    # special case phase 2 version string >= 2.0
+    return '2.040'
+
+  # sometimes the fontRevision and version string don't match, and the
+  # fontRevision is bad, so we prefer the version string.
+
+  version_string = font_data.font_version(font)
+  m = version_re.match(version_string)
+  if not m:
+    raise Exception('could not match "%s"' % version_string)
+  revision = m.group(1)
+  if float(revision) < 2.0:
+    return '2.000'
+
+  original_revision = revision
+  chars = [cp for cp in revision]
+  index = len(chars) - 1
+  while index >= 0 and chars[index] != '.':
+    if chars[index] < '9':
+      chars[index] = chr(ord(revision[index]) + 1)
+      return ''.join(chars)
+    else:
+      chars[index] = '0'
+    index -= 1
+
+  raise Exception('unable to increment revision "%s"' % original_revision)
+
+
+def _get_font_info(f):
+  font_info = noto_fonts.get_noto_font(f)
+  if not font_info:
+    raise Exception('not a noto font: "%s"' % f)
+  return font_info
+
+def _is_ui_metrics(f):
+  return _get_font_info(f).is_UI_metrics
+
+
+def _autohint_code(f, script):
+  """Return 'not-hinted' if we don't hint this, else return the ttfautohint
+  code, which might be None if ttfautohint doesn't support the script.
+  Note that LGC and MONO return None."""
```

Comment:
Oops, thanks.  I came up with no-script after writing this but I forgot to back and change it.